engine-comparison/dispatcher.sh

```diff
@@ -5,61 +5,96 @@
 . $(dirname $0)/../common.sh
 . ${SCRIPT_DIR}/common-harness.sh
 
+
 build_engine() {
-  # [[ ! -e ~/$FENGINE_CONFIGS_DIR/$1 ]] && echo "cant do" && exit 1
-  # All of this code will probably be in a separate script
 
   FENGINE_CONFIG=$1
-
+  [[ ! -e $FENGINE_CONFIG ]] && echo "File DNE" && exit 1
   echo "Building $FENGINE_CONFIG"
   FENGINE_NAME=$(basename $FENGINE_CONFIG)
-  ENGINE_DIR=$WORK/fuzz-engines/${FENGINE_NAME}
-  rm -rf $ENGINE_DIR
-  mkdir $ENGINE_DIR
-
-  # Fuzzing activities go here
+  FENGINE_DIR=$WORK/fengine-builds/${FENGINE_NAME}
+  rm -rf $FENGINE_DIR
+  mkdir $FENGINE_DIR
+
+  . $FENGINE_CONFIG
+  # Build either engine
+
+  if [[ $FUZZING_ENGINE == "libfuzzer" ]]; then
+    echo "Making a version of libfuzzer"
+    svn co http://llvm.org/svn/llvm-project/llvm/trunk/lib/Fuzzer $FENGINE_DIR
+    export LIBFUZZER_SRC=$FENGINE_DIR
+  fi
+
+  if [[ $FUZZING_ENGINE == "afl" ]]; then
+
+    # [[ ! -d $LIBFUZZER_SRC ]] && echo "Can't do AFL before libfuzzer" && break
+    if [[ ! -d $LIBFUZZER_SRC ]]; then
```

Comment:
The crux of the issue is that AFL building requires the file `afl_driver.cpp`, but this file is in the LibFuzzer repo.

The reason for this is that AFL and libfuzzer use different formats for their target files, and all of the targets that we use were designed for libfuzzer; `afl_driver.cpp` is then provided to transform these targets and make them usable with AFL.

We don't have this check for libfuzzer because the targets are compatible with libfuzzer, so this complication doesn't arise.

To view the details of using `afl_driver.cpp`, you might inspect the functions `build_libfuzzer` and `build_afl` in `common.sh`.